### PR TITLE
Final cherry pick integration

### PR DESCRIPTION
**Pull request by puerco**
#### Summary

This commit finalizes the integration of the cherrypicker code with the new git and GitHub modules. It is now working. Still to do are the mocks and integration tests but the first unit tests are already in.

#### Ticket Link
Part of: https://mattermost.atlassian.net/browse/DOPS-685

### CODE REVIEW

File: pkg/cherrypicker/cherrypicker.go [hunk 1]
```diff
@@ -156,62 +167,51 @@ func (cp *CherryPicker) CreateCherryPickPRWithContext(ctx context.Context, prNum
...
-
-		cpError = cp.impl.cherrypickCommits(
-			&cp.state, cp.options, rebaseCommits, branch,
-		)
-	}
-
-	if cpError != nil {
-		return errors.Errorf("while cherrypicking pull request %d of type %s", pr.Number, mergeMode)
 	}
+	logrus.Infof("Aqui haria push de %s ", featureBranch)
```
Comment (cpanato) [inline pkg/cherrypicker/cherrypicker.go:202]: this is correct? :)

File: pkg/git/repository.go [hunk 2]
```diff
@@ -168,18 +190,26 @@ func (di *defaultRepositoryImpl) cherryPickMergeCommit(
...
-		},
-	); err != nil {
-		return errors.Wrapf(err, "checking out %s", refName)
+	logrus.Infof("Checking out branch %s", refName)
+	// Switch to the sourceBranch, this ensures it exists and from there we branch
+	if err := command.NewWithWorkDir(
+		opts.Path, gitCommand, "checkout", refName).RunSilentSuccess(); err != nil {
+		return errors.Wrapf(err, "switching to source branch %s", refName)
 	}
+	/*
```
Comment (cpanato) [inline pkg/git/repository.go:199]: do we need to keep this? if so maybe add a more descriptive comment

File: pkg/git/repository.go [hunk 3]
```diff
@@ -200,3 +230,17 @@ func (di *defaultRepositoryImpl) pushBranch(
 	}
 	return nil
 }
+
+// func
```
Comment (cpanato) [inline pkg/git/repository.go:234]: maybe update this as well

**Review by cpanato — CHANGES_REQUESTED** [review]
just a few nits but look great! thanks for that

File: pkg/git/repository.go (hunk 2 above)
Comment (puerco) [reply pkg/git/repository.go:199]: I wanted to write this one in pure go but it turned out very problematic. I added the bash version a single commit we can revert in the future to continue the previous version

File: pkg/git/repository.go (hunk 3 above)
Comment (puerco) [reply pkg/git/repository.go:234]: Same thing as the first. Gone now.

**Comment by puerco** [comment]
Nits, done. THis will be iterating quite a lot in the next few week :+1:

**Review by cpanato — APPROVED** [review]
great work!

**Pull request merged**